voxblox/include/voxblox/integrator/integrator_utils.h

```diff
@@ -12,70 +12,89 @@
 
 namespace voxblox {
 
-// This function assumes PRE-SCALED coordinates, where one unit = one voxel
+// This class assumes PRE-SCALED coordinates, where one unit = one voxel
 // size. The indices are also returned in this scales coordinate system, which
 // should map to Local/Voxel indices.
-inline void castRay(
-    const Point& start_scaled, const Point& end_scaled,
-    std::vector<AnyIndex, Eigen::aligned_allocator<AnyIndex> >* indices) {
-  CHECK_NOTNULL(indices);
-
-  constexpr FloatingPoint kTolerance = 1e-6;
-
-  AnyIndex start_index = getGridIndexFromPoint(start_scaled);
-  AnyIndex end_index = getGridIndexFromPoint(end_scaled);
-
-  Ray ray_scaled = end_scaled - start_scaled;
-
-  AnyIndex ray_step_signs(signum(ray_scaled.x()), signum(ray_scaled.y()),
-                          signum(ray_scaled.z()));
-
-  AnyIndex corrected_step(std::max(0, ray_step_signs.x()),
-                          std::max(0, ray_step_signs.y()),
-                          std::max(0, ray_step_signs.z()));
-
-  Point start_scaled_shifted = start_scaled - start_index.cast<FloatingPoint>();
+class RayCaster {
```

Comment:
Just tested and seems pretty much the same

Original: 415495 ray casts in 00.749910 seconds
Iterative: 421508 ray casts in 00.746322 seconds